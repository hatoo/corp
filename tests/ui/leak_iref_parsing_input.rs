use futures::FutureExt;
use stap::{Cursor, Input, ParsingInput};

fn main() {
    let input = Input::new(Cursor {
        buf: Vec::<u8>::new(),
        index: 0,
    });

    let mut leak = None;

    let mut pi = ParsingInput::new(Box::new(input), |iref| {
        let leak_mut = &mut leak;
        async move {
            *leak_mut = Some(iref);
        }
        .boxed_local()
    });

    pi.poll();

    drop(leak);
}
