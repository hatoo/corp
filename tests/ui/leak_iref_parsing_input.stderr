error[E0505]: cannot move out of `leak` because it is borrowed
  --> tests/ui/leak_iref_parsing_input.rs:24:10
   |
14 |     pi.start_parsing(|iref| {
   |                      ------ borrow of `leak` occurs here
15 |         let leak_mut = &mut leak;
   |                             ---- borrow occurs due to use in closure
...
24 |     drop(leak);
   |          ^^^^ move out of `leak` occurs here
25 | }
   | - borrow might be used here, when `pi` is dropped and runs the destructor for type `ParsingInput<u8, Pin<Box<dyn futures::Future<Output = ()>>>, ()>`
